specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -283,40 +347,127 @@ not required to upload.
+  @doc("Maximum response time of the API/endpoint")
+  maxResponseTime?: int32;
+}
+
+@doc("Url test config")
+model UrlTestConfig {
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Number of steps in the ramp - up phase")
+  rampUpSteps?: int32;  
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Time to maintain a higher load after a spike in traffic")
+  spikeHoldLoad?: int32;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Multiplier applied to the base load during a spike in traffic")
+  spikeMultiplier?: float32;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Type of load. e.g. Linear, Constant, StepUp, Spike")
+  loadType?: LoadType;    
 }
 
 @doc("Optional load test config")
 model OptionalLoadTestConfig {
-  @doc("""
-Test URL. Provide the complete HTTP URL. For example,
-http://contoso-app.azurewebsites.net/login
-""")
-  endpointUrl?: string;
+  ...UrlTestConfig;
+
+  ...QuickTestConfig;
 
   @doc("No of concurrent virtual users")
   virtualUsers?: int32;
 
   @doc("Ramp up time")
-  rampUpTime?: int32;
+  rampUpTime?: int32;   
 
   @doc("Test run duration")
-  duration?: int32;
+  duration?: int32;     
 }
 
+// @doc("Optional load test config")
+// model OptionalLoadTestConfig {
+//   @doc("""
+// Test URL. Provide the complete HTTP URL. For example,
+// http://contoso-app.azurewebsites.net/login
+// """)
+//   endpointUrl?: string;
+
+//   @doc("No of concurrent virtual users")
+//   virtualUsers?: int32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Throughput required (requests per second)")
+//   rps?: int32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Maximum response time of the API/endpoint")
+//   maxResponseTime?: int32;
+
+//   @doc("Ramp up time")
+//   rampUpTime?: int32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Number of steps in the ramp - up phase")
+//   rampUpSteps?: int32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Time to maintain a higher load after a spike in traffic")
+//   spikeHoldLoad?: int32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Multiplier applied to the base load during a spike in traffic")
+//   spikeMultiplier?: float32;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Type of load. e.g. Linear, Constant, StepUp, Spike")
+//   loadType?: LoadType;
+
+//   @added(APIVersions.v2023_04_01_preview)
+//   @doc("Type of load. e.g. Linear, Constant, StepUp, Spike")
+//   testType?: TestType;
+
+//   @doc("Test run duration")
+//   duration?: int32;
+// }
+
+
+
 @doc("The input artifacts for the test.")
 model TestInputArtifacts {
   @doc("File info")
   configFileInfo?: FileInfo;
 
   @doc("File info")
   testScriptFileInfo?: FileInfo;
-
+ 
   @doc("File info")
   userPropFileInfo?: FileInfo;
 
   @doc("File info")
   inputArtifactsZipFileInfo?: FileInfo;
 
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("The config json file for url based test")
+  urlTestsConfigFileInfo?: FileInfo;
```

Comment:
This would depend on how Purva has implemented this. But since this contains the file info related to the test this would be in TestInputArtifacts in the response body.

Reply:
Let us not tie API with implementation. This is more attached to URLBasedTest settings than the test as such IMO

Reply:
The config provided in UrlBasedTest becomes InputArtifact for that test, hence I think it should be under InputArtifact for that test. This file is not provided by user but is created by us in backend based on the input provided by user AFAIK.